src/os/bluestore/BlueFS.cc

```diff
@@ -1928,24 +1929,36 @@ int BlueFS::_fsync(FileWriter *h, std::unique_lock<std::mutex>& l)
 
 void BlueFS::_flush_bdev_safely(FileWriter *h)
 {
+  std::array<bool, MAX_BDEV> flush_devs = h->dirty_devs;
+  h->dirty_devs.fill(false);
```

Comment:
I think this is racy: you could have a competing thread trying to mark it dev dirty set the value to false after we copy it but before we zero it.

Possible fix would be a second look at h->dirty_devs that does flush_devs[i] ||= h->dirty_devs[i]; ?

Reply:
I don't understand which cause racy.  Set false in lock scope.

Reply:
Oh, right!